src/plugins/janus_videoroom.c

```diff
@@ -13421,7 +13428,7 @@ static void *janus_videoroom_remote_publisher_thread(void *user_data) {
 	}
 	g_list_free(subscribers);
 	/* Free streams */
-	g_list_free(publisher->streams);
+	g_list_free_full(publisher->streams, (GDestroyNotify)(janus_videoroom_publisher_stream_unref));
```

Comment:
Other places use `janus_videoroom_publisher_stream_destroy` but as this is also used for `streams_byid`, the reference would not be decreased again, resulting in a leak.

Reply:
Not sure if this should be a destroy, then, and the related maps both be unrefs. Besides, if that's the rationale for the list, then it should not only be fixed for remote publishers, but for all of them, since we do a simple `g_list_free(participant->streams)` in `janus_videoroom_hangup_media_internal` too. That said, we should thread carefully here, as I don't remember seeing leaks normally (all my local builds use libasan).

Reply:
I did some debugging and it looks like that at the time `janus_videoroom_subscriber_free` or `janus_videoroom_publisher_free` are called, the `streams` list and the `streams_byid` / `streams_bymid` hashmaps are empty - at least in the scenarios I tested.

However at the end of the remove publisher thread, the streams list is not empty, resulting in the leak if `janus_videoroom_publisher_stream_destroy` is used.